nuxeo-services/nuxeo-invite/src/main/java/org/nuxeo/ecm/user/invite/DefaultInvitationUserFactory.java

```diff
@@ -53,10 +57,16 @@ public NuxeoPrincipal doCreateUser(CoreSession session, DocumentModel registrati
         String login = (String) registrationDoc.getPropertyValue(configuration.getUserInfoUsernameField());
         NuxeoPrincipal user = userManager.getPrincipal(login);
         if (user == null) {
+
+            if (!isSameTenant(registrationDoc, configuration)) {
+                throw new UserRegistrationException("Can only invite in same tenant");
+            }
+
+            List<String> groups = filterGroups(registrationDoc, configuration);
+
             DocumentModel newUserDoc = userManager.getBareUserModel();
             newUserDoc.setPropertyValue(UserConfig.USERNAME_COLUMN, login);
-            newUserDoc.setPropertyValue(UserConfig.PASSWORD_COLUMN,
-                    registrationDoc.getContextData(PASSWORD_KEY));
+            newUserDoc.setPropertyValue(UserConfig.PASSWORD_COLUMN, registrationDoc.getContextData(PASSWORD_KEY));
```

Comment:
Please avoid reformatting (or reorganizing imports) for code that will have to be backported.